Content.Server/GameObjects/Components/Observer/GhostComponent.cs

```diff
@@ -54,14 +54,17 @@ public override void HandleMessage(ComponentMessage message, IComponent? compone
         {
             base.HandleMessage(message, component);
 
+            EyeComponent eyeComp;
             switch (message)
             {
                 case PlayerAttachedMsg msg:
-                    msg.NewPlayer.VisibilityMask |= (int) VisibilityFlags.Ghost;
+                    eyeComp = msg.NewPlayer.AttachedEntity!.GetComponent<EyeComponent>();
```

Comment:
Should this be like an EnsureComponent instead?

Reply:
Yes.